cache/key_value_cache.go

```diff
@@ -120,3 +121,16 @@ func (k *KeyValCache) Keys(ctx context.Context, key string) ([]string, error) {
 func (k *KeyValCache) HealthCheck(ctx context.Context) error {
 	return k.redisClient.Ping(ctx).Err()
 }
+
+// Scan returns a map of keys that match the pattern
+func (k *KeyValCache) Scan(ctx context.Context, key string) (map[string]string, error) {
+	scan := make(map[string]string)
+	iter := k.redisClient.Scan(ctx, 0, "*"+key+"*", 0).Iterator()
+	for iter.Next(ctx) {
+		scan[iter.Val()] = ""
+	}
+	if err := iter.Err(); err != nil {
+		panic(err)
```

Comment:
We should probably make some sort of ErrCacheScan error type and return it here with this error wrapped rather than panicing